// Copyright 2025 The TensorFlow Quantum Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package tfq.proto;

// Store the sum of simpler terms.
message PauliSum {
  repeated PauliTerm terms = 1;
}

// Store a term which is a coefficient of some number of tensored up paulis.
message PauliTerm {
  float coefficient_real = 1;
  float coefficient_imag = 2;
  repeated PauliQubitPair paulis = 3;
}

// Store a pauli acting on a particular qubit.
message PauliQubitPair {
  string qubit_id = 1;
  string pauli_type = 2;
}
